user-guide/Advanced_Modules/DOM/DOM_examples/DOM_Altering_values_of_a_DomInstance.md

```diff
@@ -0,0 +1,250 @@
+- *void* **AddOrUpdateListFieldValue&lt;T&gt;** (SectionDefinitionID, FieldDescriptorID, T)
+
+Get:
+- *ValueWrapper&lt;T&gt;* **GetFieldValue&lt;T&gt;** (SectionDefinition, FieldDescriptor)
+- *ValueWrapper&lt;T&gt;* **GetFieldValue&lt;T&gt;** (SectionDefinitionID, FieldDescriptorID)
+- *ListValueWrapper&lt;T&gt;* **GetListFieldValue&lt;T&gt;** (SectionDefinition, FieldDescriptor)
+- *ListValueWrapper&lt;T&gt;* **GetListFieldValue&lt;T&gt;** (SectionDefinitionID, FieldDescriptorID)
+
+
+### Examples
+
+#### Add & Update
+```csharp
+// Define the ID of the DomDefinition, the SectionDefinition and the ID for one of its FieldDescriptors.
+var domDefinitionId = new DomDefinitionId(Guid.Parse("f724ff26-877e-4511-80c1-b97d17d80f79"));
+var sectionDefinitionId = new SectionDefinitionID(Guid.Parse("f45033cc-ff33-4d33-88bf-49b85e91d59c"));
+var fieldDescriptorId = new FieldDescriptorID(Guid.Parse("c97b2989-7043-4b73-a279-ab4464b0f3d6"));
+
+// Instantiate a new DomInstance & assign 'Hello' as a value for the c97b... FieldDescriptor.
+var domInstance = new DomInstance() { DomDefinitionId = domDefinitionId };
+domInstance.AddOrUpdateFieldValue<string>(sectionDefinitionId, fieldDescriptorId, "Hello");
+
+// Save it to the DB.
+var createdDomInstance = domHelper.DomInstances.Create(domInstance);
+
+// Alter the value again, you can use the same method. It will just replace the previous value.
+createdDomInstance.AddOrUpdateFieldValue<string>(sectionDefinitionId, fieldDescriptorId, "Who is this?");
+
+// Save the update to the DB.
+var updatedDomInstance = domHelper.DomInstances.Update(createdDomInstance);
+```
+
+We can also add or update list values.
+
+```csharp
+// Define the ID of a FieldDescriptor that accepts list values.
+var listFieldDescriptorId = new FieldDescriptorID(Guid.Parse("8adfb099-ac06-449a-96d0-9e652669444d"));
+
+var doubleList = new List<double>() { 26.23, 195.23, 987.42  };
+updatedDomInstance.AddOrUpdateListFieldValue<double>(sectionDefinitionId, listFieldDescriptorId, doubleList);
+
+// Save the update to the DB.
+domHelper.DomInstances.Update(updatedDomInstance);
+```
+
+#### Get
+
+```csharp
+// Retrieve the value from the DomInstance.
+var value = domInstance.GetFieldValue<string>(sectionDefinitionId, fieldDescriptorId);
+string actualString = value.Value;
+
+// Retrieve the list value from the DomInstance.
+var listValue = domInstance.GetListFieldValue<double>(sectionDefinitionId, listFieldDescriptorId);
+List<double> actualList = listValue.Values;
+```
+
+
+## Manual altering Sections and FieldValues
+
+It is also possible to manually add `Sections` and `FieldValues` to the `DomInstance`. This could be required when you, for example, want to add multiple `Sections` for the same `SectionDefinition`.
+
+### Adding a new Section & FieldValue
+
+```csharp
+// Define the ID of the DomDefinition, the SectionDefinition and the ID for one of its FieldDescriptors.
```

Comment:
@Skyline-ThomasGH Are these code blocks examples of e.g. adding a new `Section` and `FieldValue`? If so, I will add an "###Examples" subtitle, similar to the previous section on simple extension methods.

Reply:
Yes indeed. Thanks.